src/cbdata.h

```diff
@@ -269,23 +269,44 @@ cbdata_type cbdataInternalAddType(cbdata_type type, const char *label, int size)
        static cbdata_type CBDATA_##type;
 
 /// Starts cbdata-protection in a class hierarchy.
-/// Child classes in the same hierarchy should use CBDATA_CHILD().
+/// Intermediate classes in the same hierarchy must use CBDATA_INTERMEDIATE() if
+/// they risk creating cbdata pointers in their constructors.
+/// Final classes in the same hierarchy must use CBDATA_CHILD().
 class CbdataParent
 {
 public:
     virtual ~CbdataParent() {}
     virtual void *toCbdata() = 0;
+
+private:
+    /// hack: ensure CBDATA_CHILD() after a toCbdata()-defining CBDATA_INTERMEDIATE()
+    virtual void finalizedInCbdataChild() = 0;
 };
 
 /// cbdata-enables a stand-alone class that is not a CbdataParent child
 /// sets the class declaration section to "private"
 /// use this at the start of your class declaration for consistency sake
 #define CBDATA_CLASS(type) CBDATA_DECL_(type, noexcept)
 
-/// cbdata-enables a CbdataParent child class (including grandchildren)
+/// cbdata-enables a final CbdataParent-derived class in a hierarchy
 /// sets the class declaration section to "private"
 /// use this at the start of your class declaration for consistency sake
-#define CBDATA_CHILD(type) CBDATA_DECL_(type, override final)
+#define CBDATA_CHILD(type) CBDATA_DECL_(type, final) \
```

Comment:
I removed the `override` specifier here because [C.128](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#c128-virtual-functions-should-specify-exactly-one-of-virtual-override-or-final) (which I support and based this PR on) says "exactly one". I did not find any other cases where the official code combines these two specifiers.